crates/boringssl-src/src/lib.rs

```diff
@@ -0,0 +1,174 @@
+            .arg(&options.git_repo)
+            .arg(dest.as_ref().to_str().unwrap())
+            .status()?,
+        GitRef::Commit(commit_id) => {
+            Command::new("git")
+                .arg("clone")
+                .arg("--filter=tree:0")
+                .arg(&options.git_repo)
+                .arg(dest.as_ref().to_str().unwrap())
+                .status()?;
+            Command::new("git")
+                .current_dir(dest.as_ref().to_str().unwrap())
+                .arg("checkout")
+                .arg(commit_id)
+                .status()?
+        }
+    };
+
+    if !status.success() {
+        return Err(io::Error::from(ErrorKind::Other));
+    }
+
+    Ok(())
+}
+
+fn build_boringssl<P: AsRef<Path>>(dest: &P, options: &BoringSSLOptions) -> PathBuf {
+    // BoringSSL is written in C and C++, so all flags have to be given with
+    // cflags and cxxflags
+    let mut boring_conf = Config::new(&options.source_dir);
+    boring_conf
+        .define("BUILD_SHARED_LIBS", "OFF")
+        .define("CMAKE_INSTALL_PREFIX", dest.as_ref().to_str().unwrap())
+        .define("CMAKE_C_COMPILER", "clang")
+        .define("CMAKE_CXX_COMPILER", "clang++")
+        .pic(true)
+        .cflag("-g")
+        .cxxflag("-g")
+        .define("CMAKE_BUILD_TYPE", "Release")
+        .define("OPENSSL_NO_BUF_FREELISTS", "1")
+        .define("OPENSSL_NO_ASM", "1");
+
+    if options.deterministic {
+        boring_conf.define("FUZZ", "1");
+        boring_conf.define("NO_FUZZER_MODE", "1");
+    }
+
+    if options.sancov {
+        boring_conf
+            .cflag("-fsanitize-coverage=trace-pc-guard")
+            .cxxflag("-fsanitize-coverage=trace-pc-guard");
+    }
+
+    if options.gcov_analysis {
+        boring_conf
+            .cflag("-ftest-coverage")
+            .cflag("-fprofile-arcs")
+            .cxxflag("-ftest-coverage")
+            .cxxflag("-fprofile-arcs");
+    }
+
+    if options.llvm_cov_analysis {
+        boring_conf
+            .cflag("-fprofile-instr-generate")
+            .cflag("-fcoverage-mapping")
+            .cxxflag("-fprofile-instr-generate")
+            .cxxflag("-fcoverage-mapping");
+    }
+
+    if options.asan {
+        let output = Command::new("clang")
+            .args(["--print-resource-dir"])
+            .output()
+            .expect("failed to clang to get resource dir");
+        let clang: &str = std::str::from_utf8(&output.stdout).unwrap().trim();
+
+        // Important: Make sure to pass these flags to the linker invoked by rustc!
+        boring_conf
+            .cflag("-fsanitize=address")
+            .cflag("-shared-libsan")
+            .cflag("-Wno-unused-command-line-argument")
+            .cflag(format!("-Wl,-rpath={}/lib/linux/", clang))
+            .cxxflag("-fsanitize=address")
+            .cxxflag("-shared-libsan")
+            .cxxflag("-Wno-unused-command-line-argument")
+            .cxxflag(format!("-Wl,-rpath={}/lib/linux/", clang));
+    }
```

Comment:
https://github.com/tlspuffin/tlspuffin/pull/302

Reply:
Yes, we had a discussion and concluded that it's not worth spending the time to fix now. We would need to duplicate the code I used in #302 or extract it in a common crate. This will be comparatively easy to fix when we make the move to building vendor libraries outside of cargo. So I support the idea that this can remain broken on macos until then.